spec/index.md

```diff
@@ -5893,7 +5884,7 @@ Q.canister_id = ic_principal
 Q.method_name = 'fetch_canister_logs'
 Q.arg = candid(A)
 A.canister_id = effective_canister_id
-S[A.canister_id].canister_log_visibility = Public or Q.sender in S[A.canister_id].controllers or Q.sender in S[A.canister_id].canister_log_allowlist
+S[A.canister_id].canister_log_visibility = Public or (S[A.canister_id].canister_log_visibility = AllowedViewers Principals and Q.sender in Principals) or Q.sender in S[A.canister_id].controllers
```

Comment:
```suggestion
S[A.canister_id].canister_log_visibility = Public or (S[A.canister_id].canister_log_visibility = AllowedViewers Principals and Q.sender in Principals) or (S[A.canister_id].canister_log_visibility = Controllers and Q.sender in S[A.canister_id].controllers)
```
or can controllers always fetch the logs even if the log visibility is AllowedViewers and the controllers are not on the list?

Reply:
shall we allow developers to forbid fetching the logs? if yes, then allowed_viewers can be empty and should not include controllers.

Reply:
> shall we allow developers to forbid fetching the logs?

that sounds like a good motivation to apply my suggestion then; hopefully there aren't many cases when a canister has many controllers and only needs many additional log viewers (as there can be at most 10 allowed viewers, but in such a case we can bump the limit of 10 I suppose)

Reply:
> shall we allow developers to forbid fetching the logs?

We discussed in the spec meeting and in fact, one can't really prevent canister controllers from fetching the logs as they can just override the settings, fetch the logs, and set the allowed viewers back to the original list of principals. Moreover, when the set of controllers changes, the change needs to be synced in both the list of controllers and the list of allowed log viewers. So maybe it's worthwhile reconsidering the semantics of the allowed viewers' list w.r.t. canister controllers.

Reply:
@mraszyk Are you then suggesting that the semantics should be something like:

1. private -> only controllers can view
2. allowed_viewers -> controllers + principals in this list can view
3. public -> any principal is allowed to view